Lib/fontTools/subset/__init__.py

```diff
@@ -2219,16 +2229,36 @@ def subset_glyphs(self, s):
 		elif t.isUnicode():
 			t.cmap = {u:g for u,g in t.cmap.items()
 				      if g in s.glyphs_requested or u in s.unicodes_requested}
+			# Collect format 12 tables that hold only basic multilingual plane
+			# codepoints.
+			if t.format == 12 and t.cmap and max(t.cmap.keys()) < 0x10000:
+				tables_format12.append(t)
 		else:
 			t.cmap = {u:g for u,g in t.cmap.items()
 				      if g in s.glyphs_requested}
+
+	# Fomat 12 tables are redundant if they contain just the same BMP codepoints
+	# their little BMP-only encoding siblings contain.
+	for t in tables_format12:
+		if (
+			t.platformID == 0
+			and t.platEncID == 4
+			and t.language in table_plat0_enc3
+			and set(table_plat0_enc3[t.language].cmap) == set(t.cmap)
```

Comment:
why cast to set? can't you compare the dicts directly? or their .keys()?

Reply:
Works :o